c2_components/src/mfx_c2_encoder_component.cpp

```diff
@@ -1363,6 +1400,46 @@ void MfxC2EncoderComponent::DoWork(std::unique_ptr<C2Work>&& work)
             }
         }
 
+#if MFX_DEBUG_DUMP_FRAME_ENC == MFX_DEBUG_YES
+            if (MFX_IOPATTERN_IN_SYSTEM_MEMORY == m_mfxVideoParamsConfig.IOPattern &&
+	           MFX_FOURCC_NV12 == m_mfxVideoParamsState.mfx.FrameInfo.FourCC &&
+		   NeedDumpBuffer()) {
+                std::unique_ptr<const C2GraphicView> c_graph_view;
+                std::unique_ptr<C2ConstGraphicBlock> c_graph_block;
+
+                res = GetC2ConstGraphicBlock(input, &c_graph_block);
+                MapConstGraphicBlock(*c_graph_block, TIMEOUT_NS, &c_graph_view);
+
+                m_count_lock.lock();
+                if (m_count) {
+                    const uint8_t* srcY = c_graph_view->data()[C2PlanarLayout::PLANE_Y];
+                    const uint8_t* srcU = c_graph_view->data()[C2PlanarLayout::PLANE_U];
+                    const uint8_t* srcV = c_graph_view->data()[C2PlanarLayout::PLANE_V];
```

Comment:
srcV seems useless. Due to NV12 format only has two planes, the srcU can be changed to srcUV.
Also please add more comments to indicate only support NV12 format here.